File: Army(c++)/Warriors/Healer.cpp
#include "Healer.hpp"
#include "../State/State.hpp"

Healer::Healer(const std::string& name,  int hp, int mana, int dmg) 
    : SpellCaster(name, hp, mana, dmg, new SoldierState(this))
    , warrior(WarriorID::HEALER), magician(MagicianID::HEALING) {}

Healer::~Healer() {}

WarriorID Healer::getWarriorID() const {
    return this->warrior;
}

void Healer::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}

MagicianID Healer::getMagicianID() const {
	return this->magician;
}

File: Army(java)/Warriors/Healer.java
package Warriors;

import EnumID.*;
import Basic.*;

public class Healer extends SpellCaster {
    private WarriorID warrior;
    private MagicianID magician;

    public Healer(String name,  int hp, int mana, int dmg) {
        super(name, hp, mana, dmg);
        this.warrior = WarriorID.HEALER;
        this.magician = MagicianID.HEALING;
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }

    public MagicianID getMagicianID() {
        return magician;
    }
}